pcp/versions/2.0/inventory.md

```diff
@@ -8,35 +8,96 @@ Message Flow
 ---
 
 A client node that wants an inventory list of a PCP broker must 1) be
-connected to it (refer to [sessions][1]) and 2) send an inventory
-request message to a PCP broker.
+connected to it (refer to [sessions][1]) and 2) send an *inventory request*
+message to a PCP broker.
+
+If the request is processed successfully, the broker will reply with an
+*inventory response* message. Additionally, if the inventory request includes a
+`subscribe` flag set to `true`, the broker will send the client an
+*inventory update* message when the broker's inventory changes.
 
-In case the request is processed successfully, the broker will reply with an
-inventory response message.
 
 Inventory Messages
 ---
 
-Inventory request and response messages must have the *message_type*
-field respectively equal to `http://puppetlabs.com/inventory_request` and
-`http://puppetlabs.com/inventory_response`.
+PCP specifies three types of inventory messages: inventory requests,
+inventory responses, and inventory updates.
 
-The *in_reply_to* field of an inventory response contains the ID of the
-inventory request.
+For inventory requests:
+* The *message_type* field must equal `http://puppetlabs.com/inventory_request` 
+* The *data* field must conform to the following json-schema:
+```
+{
+  "properties" : {
+    "query" : { "type" : "array",
+      "items" : { "type" : "string",
+                  "pattern" : "^pcp://[^/]*/[^/]+$" }
+    },
+    "subscribe": { "type" : "boolean" }
+  },
+  "required" : ["query"],
+  "additionalProperties" : false
+}
+```
+* When provided, the *subscribe* flag must be set to `true` or `false`.
 
-The *data* field of an inventory request message is an array of URIs,
-described by the following json-schema:
+For inventory responses:
+* The *message_type* field must equal `http://puppetlabs.com/inventory_response`
+* The *in_reply_to* field must contain the ID of the inventory request.
+* The *data* field must conform to the following json-schema:
 
 ```
 {
-    "type" : "array",
-    "items" : { "type" : "string",
-                "pattern" : "^pcp://[^/]*/[^/]+$" }
+  "properties": {
+    "uris" : { "type" : "array",
+               "items" : { "type" : "string",
+                           "pattern" : "^pcp://[^/]*/[^/]+$" }
+    },
+    "version": { "type": "string",
+                 "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[0-9a-f]{4}-[0-9a-f]{12}_\d+$" },
+  },
+  "required" : ["uris", "version"],
+  "additionalProperties" : false
 }
 ```
 
-The *data* field of an inventory response message is an array of URIs,
-described by the same json-schema.
+In reference to the schema above, the `version` field is the concatenation of a
```

Comment:
I don't quite see the advantage of the UUID. Perhaps we could include the broker name and the tcp port number of the websocket connection to reasonably identify the session in the log messages. It should be possible to obtain these locally. I.e. we don't need to include the UUID in the protocol so the version can be a simple integer - which as Michael already remarked is desirable.